azure/windows-webapp/windows-webapp/variables.tf

```diff
@@ -0,0 +1,205 @@
+###################
+# Global Variables
+###################
+
+variable "location" {
+  description = "The location of created resources."
+  type        = string
+  default     = "uksouth"
+}
+
+variable "resource_group_name" {
+  description = "The name of the resource group this module will use."
+  type        = string
+  default     = null
+}
+
+variable "tags" {
+  description = "Tags applied to created resources."
+  type        = map(string)
+  default     = null
+}
+
+##############
+# App Service
+##############
+
+variable "name" {
+  description = "The name of the app service."
+  type        = string
+}
+
+variable "plan" {
+  description = "Object detailing the plan, if creating one with this module."
+  type = object({
+    create         = optional(bool, true)
+    id             = optional(string)
+    name           = optional(string)
+    sku_name       = optional(string, "B1")
+    zone_balancing = optional(bool, false)
+  })
+  default = {}
+}
+
+variable "zip_deploy_file" {
+  description = "Path to a zip file to deploy to the app service."
+  type        = string
+  default     = null
+}
+
+variable "application_stack" {
+  description = "A map detailing the application stack."
+  type        = map(string)
```

Comment:
Can we be more specific here? There is a limited set of attributes that can be set within the application_stack, if we set these in the type, they will show up in tfdocs.

This also has the added benefit of us not having to use lookup

Reply:
Kept this in line with the linux-webapp module and how that's written. May be worth factoring that in for a new PR to update both webapps as they're both similar constructs?

Reply:
Raised under https://github.com/orgs/ans-coe/projects/1/views/1?pane=issue&itemId=26670516 for webapp and functionapp modules.